account_analytic_policy/model/account_move_line.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2010 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: nhomar <nhomar@vauxoo.com>
+############################################################################
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    analytic_policy = fields.Selection(
+        [('optional', 'Optional'), ('always', 'Always'), ('never', 'Never')],
+        default='optional',
+        help='"Optional": , the accountant is free to put an analytic account '
+             'on an account move line;\n'
+             '"Always": the accountant will get an error message if there is '
+             'no analytic account;\n'
+             '"Never": the accountant will get an error message if an '
+             'analytic account is present.'
+    )
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def _is_policy_ok(self, vals):
```

Comment:
I suggest the name of the method to be changed to `_is_analytic_policy_ok`

In the future there could be other policies to be implemented on Journal Items then the name could clash with this one.

In the order hand, as your are the first one to name the method and to the implement the policy you have the right to name as you wish, other just have to adapt to use other names.

http://www.funtrivia.com/askft/Question122991.html